openapi/workflow_execution_service.openapi.yaml

```
@@ -942,8 +1122,18 @@ components:
           type: array
           items:
             type: string
-          description: An array of one or more acceptable engines versions for the `workflow_engine`
+          description: An array of one or more acceptable engines versions for the `workflow_engine`. Since a server may support multiple engines and version, the recommendation is to encode this array as `<workflow_engine>_<version>` for clarity.
       description: Available workflow engine versions supported by a given instance of the service.
+    WorkflowEngine:
+      title: WorkflowEngine
+      type: object
+      properties:
+        workflow_engine:
+          type: array
+          items:
+            type: string
+          description: An array of one or more acceptable workflow engines. Since a server may support multiple engines and version, the recommendation is to encode the workflow_engine_version array as `<workflow_engine>_<version>` where `workflow_engine` values match this array for clarity.
```

```suggestion
          description: An array of one or more acceptable workflow engines. If the server supports multiple engine versions, encode the versions in `workflow_engine_versions`.
```

1. Better not repeat the documentation of `workflow_engine_version` here, because it distracts from the meaning of `workflow_engine` and is harder to maintain (DRY principle).
2. I still think that the reference to `workflow_engine_version` is useful.